wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Civi;
+
+use Civi\Core\Service\AutoSubscriber;
+use CRM_Pets_ExtensionUtil as E;
+
+class CollectionCampHelper extends AutoSubscriber {
+
+	public static function getSubscribedEvents() {
+		return [
+			'&hook_civicrm_post' => 'generateCollectionCampCode',
+		];
+	}
+
+	/**
+	 * This hook is called after a db write on entities.
+	 *
+	 * @param string $op
+	 *   The type of operation being performed.
+	 * @param string $objectName
+	 *   The name of the object.
+	 * @param int $objectId
+	 *   The unique identifier for the object.
+	 * @param object $objectRef
+	 *   The reference to the object.
+	 */
+	public static function generateCollectionCampCode(string $op, string $objectName, int $objectId, &$objectRef) {
+		// Check if the object name is 'AfformSubmission'
+		if ($objectName !== 'AfformSubmission') {
+			return;
+		}
+
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+
+		// Access the subtype
+		$subtype = $decodedData['Eck_Collection_Camp1'][0]['fields']['subtype'] ?? null;
```

Comment:
What is the data structure of $decodedData?

I think we can do a better check here.

Reply:
when I log the decodedData it gives me this 
```
[26-Aug-2024 12:44:32 Asia/Kolkata] decodedData: Array
(
    [extra] => Array
        (
        )

    [Eck_Collection_Camp1] => Array
        (
            [0] => Array
                (
                    [fields] => Array
                        (
                            [id] => 48
                            [Collection_Camp_Core_Details.Contact_Id] => 8256
                            [Dropping_Centre.Location] => fsda
                            [Dropping_Centre.State] => 1098
                            [Collection_Camp_Core_Details.Status] => 1
                            [title] => Dropping Center
                            [subtype] => 5
                            [Collection_Camp_Core_Details.Authorizer] => 
                            [Collection_Camp_Core_Details.Authorization_date] => now
                        )

                    [joins] => Array
                        (
                        )

                )

        )

)
```

Reply:
So, if you notice, when creating a form with FormBuilder you can add multiple collection camps by adding more tabs from the UI. Now if we hardcode `[0]` then only one of the collection camp will be saved. Although we don't have a usecase for this right now but we don't want to write code in a way that limits. Instead it should be written in way that works.